### DISCUSSION TITLE
Do I need to shut down running jails before shutting down the host?

### DISCUSSION THREAD

**Discussion by Kailee71** [Jip-Hop/jailmaker#64] [Q&A]
Hi all,

jailmaker seems to work very well in my environment (Truenas 23.10), and am very happy with installation/setup/networking/admin. I do have one question though - automatic startup of jails work very also, but do I have to manually shut down the jails before I shut down the host to prevent any damage to the jail? Or is there some way of automating jail shutdown with a shutdown script in TrueNAS? I ask because

`jlmkr startup`

works for all available jails, but there is no equivalent to stop all running jails.

Any hints very welcome :-)

Kai.

**Comment by Jip-Hop** [comment]
Glad to hear you like jailmaker. There's no need to worry about manual shutdown. This is handled automatically due to standard behavior of systemd-nspawn: https://www.freedesktop.org/software/systemd/man/latest/systemd-nspawn.html#--kill-signal=
